pkg/apis/sensor/v1alpha1/types.go

```diff
@@ -688,6 +691,36 @@ type CustomTrigger struct {
 	Payload []TriggerParameter `json:"payload" protobuf:"bytes,7,rep,name=payload"`
 }
 
+// EmailTrigger refers to the specification of the email notification trigger.
+type EmailTrigger struct {
+	// Parameters is the list of key-value extracted from event's payload that are applied to
+	// the trigger resource.
+	// +optional
+	Parameters []TriggerParameter `json:"parameters,omitempty" protobuf:"bytes,1,rep,name=parameters"`
+	// Username refers to the username used to connect to the smtp server.
+	Username string `json:"username,omitempty" protobuf:"bytes,2,opt,name=username"`
+	// SMTPPassword refers to the Kubernetes secret that holds the smtp password used to connect to smtp server.
+	SMTPPassword *corev1.SecretKeySelector `json:"smtpPassword,omitempty" protobuf:"bytes,3,opt,name=smtpPassword"`
+	// Host refers to the smtp host url to which email is send.
+	Host string `json:"host,omitempty" protobuf:"bytes,4,opt,name=host"`
+	// Port refers to the smtp server port to which email is send.
+	// Defaults to 0.
+	// +optional
+	Port int32 `json:"port,omitempty" protobuf:"varint,5,opt,name=port"`
+	// To refers to the email address to which email is send.
+	// +optional
+	To string `json:"to,omitempty" protobuf:"bytes,6,opt,name=to"`
```

Comment:
Is making it `[]string` better?

Reply:
Making it ```[] string``` would better. But  argoproj/notifications-engine doesn't support multiple to emails.
This line passes only one to address down.
https://github.com/argoproj/notifications-engine/blob/master/pkg/services/email.go#L84C43-L84C43